### PR TITLE
docs: Use Secrets to create superusers in the Helm chart

### LINKED ISSUE #1106: Configure SASL for Redpanda in Kubernetes
### Describe the Issue

Include the ability to reference Secrets for the passwords as described in: https://github.com/redpanda-data/helm-charts/pull/295

### Updates to existing documentation

https://docs.redpanda.com/docs/manage/kubernetes/security/sasl-kubernetes/

### New feature or content gap requests

Content gap.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: February 15

Fixes #1106 

Open questions:

- Does the Helm chart support Kerberos? Answer: not at the moment.
- Does the Helm chart support configuring SASL for specific listeners, or is it configured for all listeners by default? Answer: enabled for Schema Registry and HTTP proxy when SASL is enabled

### CODE REVIEW

**Review by RafalKorepta — APPROVED** [review]
LGTM

File: docs/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 1]
```diff
@@ -10,42 +10,142 @@ tags:
...
+  Replace the following placeholders with your own values for the superuser:
 
-```yaml title="sasl-enable.yaml"
+    - `<superuser-name>`
+    - `<superuser-password>`
+    - `<superuser-authentication-mechanism>`
+
+  ```bash
+  echo '<superuser-name>:<superuser-password>:<superuser-authentication-mechanism>' > superuser.txt
+  ```
```
Comment (shorgi) [inline docs/manage/kubernetes/security/sasl-kubernetes.mdx:56]: Would this command look better before or after the `Replace the following placeholders ...` list? I prefer the command first, then the clarification. In the current preview, the indentation of the code-block after the bullets is correct, but still looks off.

File: docs/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 2]
```diff
@@ -57,48 +157,63 @@ kubectl exec -n redpanda -c redpanda redpanda-0 -- \
...
     rpk acl create --allow-principal User:myuser \
     --operation describe \
     --topic myfirsttopic \
-    --user admin --password changeme --sasl-mechanism SCRAM-SHA-256
+    --user <superuser-name> --password <superuser-password> --sasl-mechanism <superuser-authentication-mechanism>
   ```
 
 :::note
-If a user has `describe` privileges on a cluster, it does not mean that user is automatically granted `describe` privileges on topics.
+If a user has `describe` privileges for a cluster, it does not mean that user is automatically granted `describe` privileges for topics.
```
Comment (shorgi) [inline docs/manage/kubernetes/security/sasl-kubernetes.mdx:189]:
```suggestion
`describe` privileges for topics must be explicitly granted to a user. Even if a user has `describe` privileges for a cluster, it does not mean that user is automatically granted `describe` privileges for topics. 
```

**Review by shorgi — APPROVED** [review]
Looks good, the comments also apply to v22.3.

File: docs/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 3]
```diff
@@ -7,45 +7,149 @@ tags:
...
+
+To use a Secret to store superuser credentials, do the following:
+
+1. Create a file in which to store the credentials:
+
+  Replace the following placeholders with your own values for the superuser:
+
+    - `<superuser-name>`
+    - `<superuser-password>`
+    - `<superuser-authentication-mechanism>`
```
Comment (JakeSCahill) [inline docs/manage/kubernetes/security/sasl-kubernetes.mdx:56]:
```suggestion
    - `<superuser-authentication-mechanism>` (SCRAM-SHA-256 or SCRAM-SHA-512)
```

**Comment by JakeSCahill** [comment]
@alejandroEsc if we use a secret, must we set the `auth.sasl.users` to an empty object?

I created an empty secret called `redpanda-users` and kept the `admin` user in the list, then Helm complains that the secret already exists. I assume if a list of users is given, the Helm chart creates a secret with the given name in `secretRef`?

File: docs/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 4]
```diff
@@ -57,48 +161,63 @@ kubectl exec -n redpanda -c redpanda redpanda-0 -- \
...
 
 To describe the topic:
 
 ```bash
 kubectl exec -n redpanda -c redpanda <cluster_name>-0 -- \
   rpk topic describe myfirsttopic \
-  --user myuser --password changethispassword --sasl-mechanism SCRAM-SHA-256
+  --user myuser --password changethispassword --sasl-mechanism <superuser-authentication-mechanism>
 ```
+
```
Comment (alejandroEsc) [inline docs/manage/kubernetes/security/sasl-kubernetes.mdx:215]: Probably a tip section indicating that the --sasl-mechanism is required.

File: versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 5]
```diff
@@ -10,42 +10,142 @@ tags:
...
 </head>
 
-Simple Authentication and Security Layer (SASL) is a method for adding authentication support to connection-based protocols. When using the Redpanda Helm chart, SASL provides authentication between Redpanda brokers and clients.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ValuesYaml from '../shared/_values-yaml.mdx'
 
-:::note
-You must use [TLS encryption](../kubernetes-tls) to have secure authentication using SASL.
+SASL (Simple Authentication and Security Layer) is a method for adding authentication support to connection-based protocols. In the Redpanda Helm chart, enabling SASL provides authentication between all Redpanda brokers and clients.
```
Comment (alejandroEsc) [inline versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx:17]:
```suggestion Simple Authentication and Security Layer (SASL) is a method for adding authentication support to connection-based protocols. In the Redpanda Helm chart, enabling SASL provides authentication between all Redpanda brokers and clients.
```

File: versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 6]
```diff
@@ -10,42 +10,142 @@ tags:
...
+
+:::tip
+For secure authentication, you must use [TLS encryption](../kubernetes-tls) with SASL.
 :::
 
+## Supported authentication mechanisms
+
+The Redpanda Helm chart supports SCRAM (Salted Challenge Response Authentication Mechanism).
```
Comment (alejandroEsc) [inline versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx:25]:
```suggestion
The Redpanda Helm chart supports Salted Challenge Response Authentication Mechanism (SCRAM) .
```

File: versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx [hunk 7]
```diff
@@ -10,42 +10,142 @@ tags:
...
+
+SCRAM provides strong encryption for user names and passwords by default and does not require an external data store for user information. Redpanda supports  `SCRAM-SHA-256` and `SCRAM-SHA-512`.
```
Comment (alejandroEsc) [inline versioned_docs/version-22.3/manage/kubernetes/security/sasl-kubernetes.mdx:27]:
```suggestion
SCRAM provides strong encryption for user names and passwords by default and does not require an external data store for user information. Redpanda only supports `SCRAM-SHA-256` and `SCRAM-SHA-512`.
```

**Review by alejandroEsc — COMMENTED** [review]
Left a few suggestions. Some are regarding style, which I could totally be wrong about and thats ok.

**Review by RafalKorepta — APPROVED** [review]

**Review by alejandroEsc — APPROVED** [review]
lgtm!

**Pull request merged**